src/components/common/ErrorBoundary.tsx

```diff
@@ -0,0 +1,57 @@
+import React, {ErrorInfo} from "react";
+import styled from "styled-components";
+import colors from "consts/colors";
+
+const ErrorContainer = styled.div`
+  margin: 0 0 2rem;
+`;
+
+const ErrorTitle = styled.h2`
+  color: ${colors.white}
+`;
+
+const ErrorDetails = styled.details`
+  color: ${colors.white}
+`;
+
+interface IState {
+    error: Error,
+    errorInfo: ErrorInfo
+}
+
+interface IProps {
+
+}
+
+class ErrorBoundary extends React.Component<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+        this.state= { error: null, errorInfo: null };
+    }
+
+    componentDidCatch(error: Error, errorInfo: ErrorInfo) {
+        this.setState({
+            error: error,
+            errorInfo: errorInfo
+        })
+    }
+
+    render() {
+        if (this.state.errorInfo) {
+            return (
+                <ErrorContainer>
+                    <ErrorTitle>Whooops! Something went wrong...</ErrorTitle>
```

Comment:
Let's move all strings to separate files with constants to make them more reusable.